### PR TITLE
EOS-22641: F-11D: CSM setup should be updated to accept admin users and credentials in conf store template and create user in config phase

### PR DESCRIPTION
**Pull request by udayan-y21**
Signed-off-by: Udayan Yaragattikar <udayan.yaragattikar@seagate.com>

# Backend

# Problem Statement
https://jts.seagate.com/browse/EOS-22641
- _Overview: JIRA number/GitHub Issue added to PR: [Y/N]:_Y
- _Describe the problem this patch intends to solve._
Cluster admin creation in miniprovisioning step and Change password popup for firsttime login

## Design
1. Create cluster admin in miniprovisioning config phase.

2. Consume confstore keys in config phase. 

3. If no key present in confstore for cluster admin credentials, then default username and password will be taken

4. If admin already created and user tries to rerun the config phase then warning message to show that user already exist. Use force flag to perform action forcefully
5. If '-f' flag provided then: 
If same user is recreated then delete older user and created new one 
If another user is given then created another user.
 
6. Reset password for first time login.
  

## Coding

- _Coding conventions are followed and code is consistent. Yes/No?_Y
- _Confirm All CODACY errors are resolved. Yes/No?_Y

## Testing
image
image
image
image

## Checklist

- _PR is self-reviewed? Yes/No?_Y
- _GitHub Issue is updated_NA
- _Jira is updated_Y
  -  _Check if the description is clear and explained._Y 
    -  _Check Acceptance Criterion is defined._Y
      -  _All the tests performed should be mentioned before Resolving a JIRA._Y
        -  _Verification needs to be done before marked as Closed/Verified_Y
        - _Any interface change Yes/No?_N
        - _Change notified to other gatekeepers: Yes/No?_NA
        - _Code reviews done and addressed: Yes/No?_
        - _Codacy issues reviewed and addressed: Yes/No?_Y
        - _Deployment test : Done/Not?_Y
        - _UT/smoke test: Done/Not?_Y
        - _Jira number added to PR: Yes/No?_Y
        - _Github merge done using UI: Yes/No?_Y
        - _Side effects on other features - deployment/update/node-replacement_
        - _Changes to quick-start-guide/community impact_
        - _All dependent component code PR are raised or already merged Yes/No_
        - _All dependent code already merged in landing branch Yes/No_

### CODE REVIEW

**Review by pawankumarsrivastava — APPROVED** [review]
Add regex validation to py-utils.

File: csm/conf/configure.py [hunk 1]
```diff
@@ -256,3 +267,42 @@ def get_component_list_from_features_endpoints():
         except Exception as e_:
             Log.error(f"Error in storing unsupported features: {e_}")
             raise CsmSetupError(f"Error in storing unsupported features: {e_}")
+
+    async def _create_cluster_admin(self):
+        # TODO confstore keys can be changed.
```
Comment (kupranay) [inline csm/conf/configure.py:272]: as discussed please move this logic to usermanager class

**Review by kupranay — CHANGES_REQUESTED** [review]

**Comment by udayan-y21** [comment]
> Add regex validation to py-utils.

As per discussion with Pranay, writing regex validator is not required for now.. Coz Password validator is bit complex in CSM where validation error raises different errors for different string. Its achivable but complex at the moment. SO using csm egex for now.

File: csm/conf/configure.py (hunk 1 above)
Comment (udayan-y21) [reply csm/conf/configure.py:272]: Modifications done

**Review by kupranay — APPROVED** [review]

**Pull request merged**